e2e/testcases/reconciler_manager_test.go

```diff
@@ -772,389 +712,276 @@ func firstContainerCPURequestIs(cpuRequest int64) testpredicates.Predicate {
+			// so read, update, and write it back.
+			updated := expectedResources[firstContainerName]
+			updated.CPURequest.Add(diff)
+			expectedResources[firstContainerName] = updated
+			// Update total
+			expectedTotalResources.CPURequest = minimumTotalCPURequests
+		}
+	}
+
+	nt.T.Log("Validating container resources - 1")
+	reconcilerDeployment = &appsv1.Deployment{}
+	err = nt.Validate(reconcilerNN.Name, reconcilerNN.Namespace, reconcilerDeployment,
+		testpredicates.HasGenerationAtLeast(generation),
+		totalContainerCPURequestEquals(expectedTotalResources.CPURequest),
+		totalContainerMemoryRequestEquals(expectedTotalResources.MemoryRequest),
+		firstContainerNameEquals(firstContainerName),
+		testpredicates.DeploymentContainerResourcesAllEqual(expectedResources),
 	)
 	if err != nil {
+		nt.T.Log("Reconciler container specs (expected):")
+		for containerName, containerSpec := range expectedResources {
+			nt.T.Logf("%s: %s", containerName, log.AsJSON(containerSpec))
+		}
+		nt.T.Log("Reconciler container specs (found):")
+		for _, container := range reconcilerDeployment.Spec.Template.Spec.Containers {
+			nt.T.Logf("%s: %s", container.Name, log.AsJSON(container.Resources))
+		}
 		nt.T.Fatal(err)
 	}
-	generation = getDeploymentGeneration(nt, nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem)
-	// increase CPU and memory request to above current request autopilot increment
-	nt.T.Log("Increasing CPU and memory request to above current request autopilot increment")
-	updatedFirstContainerCPURequest := firstContainerCPURequest + 10
-	updatedFirstContainerMemoryRequest := firstContainerMemoryRequest + 100
-	nt.MustMergePatch(rs, fmt.Sprintf(`{"spec":{"override":{"resources":[{"containerName":"%s","memoryRequest":"%dMi", "cpuRequest":"%dm"}]}}}`,
-		firstContainerName, updatedFirstContainerMemoryRequest, updatedFirstContainerCPURequest))
-	// generation = 2
-	generation++
+	generation = reconcilerDeployment.GetGeneration()
+
+	nt.T.Log("Increasing CPU and Memory request on the RootSync spec.override")
+	updated := expectedResources[firstContainerName]
+	updated.CPURequest.Add(resource.MustParse("10m"))
+	updated.MemoryRequest.Add(resource.MustParse("10Mi"))
+	nt.MustMergePatch(rootSyncObj,
+		fmt.Sprintf(`{"spec":{"override":{"resources":[{"containerName":%q,"memoryRequest":%q, "cpuRequest":%q}]}}}`,
+			firstContainerName, &updated.MemoryRequest, &updated.CPURequest))
 
+	// Update expectations
+	expectedResources[firstContainerName] = updated
 	if nt.IsGKEAutopilot {
-		// hydration-controller input: 190m/200Mi
-		// with autopilot adjustment, CPU of container[0]/hydration-controller is increased to 430m
-		// bringing the total to 500m
-		expectedTotalCPU += autopilotCPUIncrements
-		expectedFirstContainerCPU = expectedFirstContainerCPU2
-		expectedFirstContainerMemory = expectedFirstContainerMemory2
+		// Round up to the nearest multiple of 250m.
+		updated := expectedResources[firstContainerName]
```

Comment:
These snippets where we calculate the expected requests/limits are a bit hard to follow IMO. Assuming the expected value is constant for standard and autopilot, can we inline these rather than doing math in the test to evaluate the expected value? I think this would help with readability and also avoids potential bugs in the test's math. I would suggest inlining the values rather than declaring a const elsewhere like before, as the variables also make it a bit hard to read. WDYT?

Reply:
> Assuming the expected value is constant for standard and autopilot, can we inline these rather than doing math in the test to evaluate the expected value?